### ISSUE TITLE
[BUG] `OpensearchIndexTemplate` always requires update

### ISSUE THREAD

**Issue by vchirikov** [opensearch-project/opensearch-k8s-operator#731]
Template:

```yaml
apiVersion: opensearch.opster.io/v1
kind: OpensearchIndexTemplate
metadata:
  name: all
  namespace: infra
spec:
  opensearchCluster:
    name: opensearch
  name: default_template
  indexPatterns:
    - "*"
  template:
    settings:
      number_of_shards: 6
      number_of_replicas: 0
```

logs:
```
{"level":"info","ts":"2024-02-20T13:36:08.036Z","msg":"Reconciling OpenSearchCluster","controller":"opensearchcluster","controllerGroup":"opensearch.opster.io","controllerKind":"OpenSearchCluster","OpenSearchCluster":{"name":"opensearch","namespace":"infra"},"namespace":"infra","name":"opensearch","reconcileID":"f0028c11-64b5-48a1-90f1-59d01a9eb1c9","cluster":{"name":"opensearch","namespace":"infra"}}
{"level":"info","ts":"2024-02-20T13:36:08.053Z","msg":"Generating certificates","controller":"opensearchcluster","controllerGroup":"opensearch.opster.io","controllerKind":"OpenSearchCluster","OpenSearchCluster":{"name":"opensearch","namespace":"infra"},"namespace":"infra","name":"opensearch","reconcileID":"f0028c11-64b5-48a1-90f1-59d01a9eb1c9","interface":"transport"}
{"level":"info","ts":"2024-02-20T13:36:08.053Z","msg":"Generating certificates","controller":"opensearchcluster","controllerGroup":"opensearch.opster.io","controllerKind":"OpenSearchCluster","OpenSearchCluster":{"name":"opensearch","namespace":"infra"},"namespace":"infra","name":"opensearch","reconcileID":"f0028c11-64b5-48a1-90f1-59d01a9eb1c9","interface":"http"}
{"level":"info","ts":"2024-02-20T13:36:29.712Z","msg":"Reconciling OpensearchIndexTemplate","controller":"opensearchindextemplate","controllerGroup":"opensearch.opster.io","controllerKind":"OpensearchIndexTemplate","OpensearchIndexTemplate":{"name":"all","namespace":"infra"},"namespace":"infra","name":"all","reconcileID":"9703a2eb-0a1a-49b8-bdac-3fe57a3bca5b","indextemplate":{"name":"all","namespace":"infra"}}
{"level":"info","ts":"2024-02-20T13:36:29.759Z","msg":"OpenSearch Index template requires update","controller":"opensearchindextemplate","controllerGroup":"opensearch.opster.io","controllerKind":"OpensearchIndexTemplate","OpensearchIndexTemplate":{"name":"all","namespace":"infra"},"namespace":"infra","name":"all","reconcileID":"9703a2eb-0a1a-49b8-bdac-3fe57a3bca5b"}
{"level":"info","ts":"2024-02-20T13:36:38.138Z","msg":"Reconciling OpenSearchCluster","controller":"opensearchcluster","controllerGroup":"opensearch.opster.io","controllerKind":"OpenSearchCluster","OpenSearchCluster":{"name":"opensearch","namespace":"infra"},"namespace":"infra","name":"opensearch","reconcileID":"c0e46c28-168f-490f-a51f-098815c94bbb","cluster":{"name":"opensearch","namespace":"infra"}}
{"level":"info","ts":"2024-02-20T13:36:38.156Z","msg":"Generating certificates","controller":"opensearchcluster","controllerGroup":"opensearch.opster.io","controllerKind":"OpenSearchCluster","OpenSearchCluster":{"name":"opensearch","namespace":"infra"},"namespace":"infra","name":"opensearch","reconcileID":"c0e46c28-168f-490f-a51f-098815c94bbb","interface":"transport"}
{"level":"info","ts":"2024-02-20T13:36:38.157Z","msg":"Generating certificates","controller":"opensearchcluster","controllerGroup":"opensearch.opster.io","controllerKind":"OpenSearchCluster","OpenSearchCluster":{"name":"opensearch","namespace":"infra"},"namespace":"infra","name":"opensearch","reconcileID":"c0e46c28-168f-490f-a51f-098815c94bbb","interface":"http"}
{"level":"info","ts":"2024-02-20T13:36:59.773Z","msg":"Reconciling OpensearchIndexTemplate","controller":"opensearchindextemplate","controllerGroup":"opensearch.opster.io","controllerKind":"OpensearchIndexTemplate","OpensearchIndexTemplate":{"name":"all","namespace":"infra"},"namespace":"infra","name":"all","reconcileID":"6642ef06-985f-43f0-9bbe-746855db23f7","indextemplate":{"name":"all","namespace":"infra"}}
{"level":"info","ts":"2024-02-20T13:36:59.809Z","msg":"OpenSearch Index template requires update","controller":"opensearchindextemplate","controllerGroup":"opensearch.opster.io","controllerKind":"OpensearchIndexTemplate","OpensearchIndexTemplate":{"name":"all","namespace":"infra"},"namespace":"infra","name":"all","reconcileID":"6642ef06-985f-43f0-9bbe-746855db23f7"}
{"level":"info","ts":"2024-02-20T13:37:08.269Z","msg":"Reconciling OpenSearchCluster","controller":"opensearchcluster","controllerGroup":"opensearch.opster.io","controllerKind":"OpenSearchCluster","OpenSearchCluster":{"name":"opensearch","namespace":"infra"},"namespace":"infra","name":"opensearch","reconcileID":"3e68522e-0141-4b27-ad36-be5de48d0412","cluster":{"name":"opensearch","namespace":"infra"}}
{"level":"info","ts":"2024-02-20T13:37:08.289Z","msg":"Generating certificates","controller":"opensearchcluster","controllerGroup":"opensearch.opster.io","controllerKind":"OpenSearchCluster","OpenSearchCluster":{"name":"opensearch","namespace":"infra"},"namespace":"infra","name":"opensearch","reconcileID":"3e68522e-0141-4b27-ad36-be5de48d0412","interface":"transport"}
{"level":"info","ts":"2024-02-20T13:37:08.290Z","msg":"Generating certificates","controller":"opensearchcluster","controllerGroup":"opensearch.opster.io","controllerKind":"OpenSearchCluster","OpenSearchCluster":{"name":"opensearch","namespace":"infra"},"namespace":"infra","name":"opensearch","reconcileID":"3e68522e-0141-4b27-ad36-be5de48d0412","interface":"http"}
```

And also I see `Generating certificates` each reconcile loop...

docker image: `opensearchproject/opensearch-operator:2.5.1`

**Comment by evheniyt** [comment]
I believe it's the same issue as in #633. I have prepared PR #727 to fix it, I will check if your issue is the same and if yes - will add a fix for `OpensearchIndexTemplate` reconciler too

**Comment by evheniyt** [comment]
@vchirikov So in your case besides the issue mentioned in #727 there are 2 other issues:
1) Opensearch API returns all settings in quotes, so to match it you need to change
```
template:
    settings:
      number_of_shards: 6
      number_of_replicas: 0
```
to
```
template:
    settings:
      number_of_shards: "6"
      number_of_replicas: "0"
```
2) There is also an issue in comparing `OpensearchIndexTemplate` object with API response from Opensearch. I have debugged it and here is what I have for my `OpensearchIndexTemplate` settings
```
{
   "index":{
      "mapping":{
         "total_fields":{
            "limit":"2000"
         }
      },
      "number_of_replicas":"1",
      "number_of_shards":"5",
      "refresh_interval":"60s",
      "unassigned":{
         "node_left":{
            "delayed_timeout":"15m"
         }
      }
   }
}
```
but API returns unsorted json object:
```
{
   "index":{
      "mapping":{
         "total_fields":{
            "limit":"2000"
         }
      },
      "refresh_interval":"60s",
      "unassigned":{
         "node_left":{
            "delayed_timeout":"15m"
         }
      },
      "number_of_shards":"5",
      "number_of_replicas":"1"
   }
}
```
because of this, the comparing function used to check the diff is always seeing a diff and trying to update the object.

I have added a fix to my #727 PR that should solve this sorting issue.
